### ISSUE TITLE
[Support]: Crashing on Migration from 0.10.1 to 0.11 or 0.11.1

### ISSUE THREAD

**Issue by joeShuff** [blakeblackshear/frigate#4148]
### Describe the problem you are having

I tried to migrate from 0.10.1 up to 0.11.1 (and after this error I also tried 0.11 but same issue) and the container isn't starting and keeps stopping/restarting itself (probably docker autorestarting) and failing on the same database migration.

Fortunately I made a backup of everything and have just reverted to 0.10.1 for the time being.

### Version

0.10.1-83481af

### Frigate config file

```yaml
logger:
  default: debug
  logs:
    peewee: info
mqtt:
  host: <ip>
  port: 1883
  topic_prefix: frigate
  client_id: frigate
  user: <user>
  password: <pass>
  stats_interval: 60
cameras:
  living_room:
    ffmpeg:
      inputs:
        - path: http://<ip>/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=<user>&password=<pass>
          roles:
            - rtmp
            - record
        - path: http://<ip>/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=<user>&password=<pass>
          roles:
            - detect
      output_args:    
        record: -f segment -segment_time 60 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy
      input_args:
        - -avoid_negative_ts
        - make_zero
        - -fflags
        - nobuffer+genpts+discardcorrupt
        - -flags
        - low_delay
        - -strict
        - experimental
        - -analyzeduration
        - 1000M
        - -probesize
        - 1000M
        - -rw_timeout
        - "5000000"
    detect:
      width: 896
      height: 672
      fps: 5
      max_disappeared: 25
    record:
      enabled: true
      retain:
        days: 90
        mode: motion
      events:
        pre_capture: 3
        post_capture: 3
        objects:
          - person
          - cat
        retain:
          default: 365
          mode: motion
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      retain:
        default: 365
    objects:
      track:
        - person
        - cat   
    motion:
      mask:
        - 896,0,896,57,500,53,504,0
  front_bedroom:
    ffmpeg:
      inputs:
        - path: http://<ip>/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=<user>&password=<pass>
          roles:
            - rtmp
            - record
        - path: http://<ip>/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=<user>&password=<pass>
          roles:
            - detect
      output_args:    
        record: -f segment -segment_time 60 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy
      input_args:
        - -avoid_negative_ts
        - make_zero
        - -fflags
        - nobuffer+genpts+discardcorrupt
        - -flags
        - low_delay
        - -strict
        - experimental
        - -analyzeduration
        - 1000M
        - -probesize
        - 1000M
        - -rw_timeout
        - "5000000"
    detect:
      width: 896
      height: 672
      fps: 5
      max_disappeared: 25
    record:
      enabled: true
      retain:
        days: 90
        mode: motion
      events:
        pre_capture: 3
        post_capture: 3
        objects:
          - person
          - cat
        retain:
          default: 365
          mode: motion
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      retain:
        default: 365
    objects:
      track:
        - person
        - cat   
    motion:
      mask:
        - 896,0,896,57,500,53,504,0
  back_bedroom:
    ffmpeg:
      inputs:
        - path: http://<ip>/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=<user>&password=<pass>
          roles:
            - rtmp
            - record
        - path: http://<ip>/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=<user>&password=<pass>
          roles:
            - detect
      output_args:    
        record: -f segment -segment_time 60 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy
      input_args:
        - -avoid_negative_ts
        - make_zero
        - -fflags
        - nobuffer+genpts+discardcorrupt
        - -flags
        - low_delay
        - -strict
        - experimental
        - -analyzeduration
        - 1000M
        - -probesize
        - 1000M
        - -rw_timeout
        - "5000000"
    detect:
      width: 896
      height: 672
      fps: 5
      max_disappeared: 25
    record:
      enabled: true
      retain:
        days: 90
        mode: motion
      events:
        pre_capture: 3
        post_capture: 3
        objects:
          - person
          - cat
        retain:
          default: 365
          mode: motion
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      retain:
        default: 365
    objects:
      filters:
        person:
          mask:
            - 618,399,687,399,729,402,801,420,825,448,896,492,896,562,896,672,840,672,741,672,683,600,650,540,615,497,543,438
      track:
        - person
        - cat   
    motion:
      mask:
        - 896,0,896,57,500,53,504,0
  feeder:
    ffmpeg:
      inputs:
        - path: http://<ip>:8081
          roles:
            - record  
            - detect
      output_args:    
        record: -f segment -segment_time 60 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c:v libx264 -an
      input_args:
        - -avoid_negative_ts
        - make_zero
        - -fflags
        - nobuffer
        - -flags
        - low_delay
        - -strict
        - experimental
        - -fflags
        - +genpts+discardcorrupt
        - -r
        - "3" # <---- adjust depending on your desired frame rate from the mjpeg image
        - -use_wallclock_as_timestamps
        - "1"
    detect:
      enabled: false
      width: 640
      height: 480
      fps: 5
      max_disappeared: 25
    record:
      enabled: true
      retain:
        days: 1
        mode: all
      events:
        pre_capture: 3
        post_capture: 3
        objects:
          - person
          - cat
        retain:
          default: 7
          mode: active_objects
    rtmp:
      enabled: false
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      retain:
        default: 365 
  doorbell:
    ffmpeg:
      inputs:
        - path: rtsp://<user>:<pass>@<ip>:554
          roles:
            - rtmp
            - record
        - path: rtsp://<user>:<pass>@<ip>:554/cam/realmonitor?channel=1&subtype=1
          roles:
            - detect
      output_args:
        record: -f segment -segment_time 60 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy
    detect:
      width: 720
      height: 576
      fps: 5      
      max_disappeared: 25
    record:
      enabled: true
      retain:
        days: 365
        mode: motion
      events:
        pre_capture: 5
        post_capture: 5
        objects:
          - person
          - cat
        retain:
          default: 365
          mode: motion
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      retain:
        default: 365
    motion:
      threshold: 15
      contour_area: 15
      mask:
        - 119,235,201,203,276,156,354,177,410,174,422,244,466,228,484,162,579,169,678,171,720,165,720,127,720,0,0,0,0,253
        - 550,576,720,576,720,500,720,442,657,429,571,437,490,477,468,529,494,561
        - 438,576,384,542,344,520,263,508,208,523,171,542,120,576
        - 0,576,122,576,122,537,111,474,73,428,0,422
    zones: 
      driveway:
        coordinates: 720,503,720,576,720,576,0,576,77,417,164,358,208,340,342,364,469,372,603,491
    objects:
      filters:
        person:
          mask:
            - 556,454,524,426,462,373,328,368,239,356,167,343,96,331,0,315,0,0,720,0,720,454,684,437
      track:
        - person
        - cat
  garden:
    ffmpeg:
      inputs:
        - path: rtsp://<user>:<pass>@<ip>:554
          roles:
            - rtmp
            - record
        - path: rtsp://<user>:<pass>@<ip>:554/cam/realmonitor?channel=1&subtype=1
          roles:
            - detect
      output_args:
        record: -f segment -segment_time 60 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy
    detect:
      width: 704
      height: 480
      fps: 5      
      max_disappeared: 25
    record:
      enabled: true
      retain:
        days: 365
        mode: motion
      events:
        pre_capture: 5
        post_capture: 5
        objects:
          - person
          - cat
        retain:
          default: 365
          mode: motion
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      retain:
        default: 365
    motion:
      mask: 
        - 704,0,704,42,704,150,661,119,599,83,565,59,513,42,512,0
        - 48,0,138,0,142,86,70,113
    objects:
      filters:
        person:
          mask:
            - 505,0,704,0,704,55,704,90,704,156,623,100,593,70,548,47
      track:
        - person
        - cat
detectors:
  coral:
    type: edgetpu
    device: usb
birdseye:
  enabled: true
  mode: continuous
  height: 1080
  width: 1920
```

### Relevant log output

```shell
Starting migrations
[2022-10-23 19:38:40] peewee_migrate                 INFO    : Starting migrations
[2022-10-23 19:38:40] frigate.app                    INFO    : Starting Frigate (0.11.0-3846a13)
Migrate "011_update_indexes"
[2022-10-23 19:38:40] peewee_migrate                 INFO    : Migrate "011_update_indexes"
sql ('CREATE INDEX "event_start_time_end_time" ON "event" ("start_time" DESC, "end_time" DESC)',)
[2022-10-23 19:38:40] peewee_migrate                 INFO    : sql ('CREATE INDEX "event_start_time_end_time" ON "event" ("start_time" DESC, "end_time" DESC)',)
sql ('DROP INDEX recordings_start_time_end_time',)
[2022-10-23 19:38:40] peewee_migrate                 INFO    : sql ('DROP INDEX recordings_start_time_end_time',)
no such index: recordings_start_time_end_time
Traceback (most recent call last):
Migration failed: 011_update_indexes
  File "/usr/local/lib/python3.9/dist-packages/peewee.py", line 3177, in execute_sql
sqlite3.OperationalError: no such index: recordings_start_time_end_time
    cursor.execute(sql, params or ())
During handling of the above exception, another exception occurred:
  File "/usr/local/lib/python3.9/dist-packages/peewee_migrate/router.py", line 179, in run_one
  File "/usr/local/lib/python3.9/dist-packages/peewee_migrate/migrator.py", line 151, in run
    migrator.run()
```

### FFprobe output from your camera

```shell
<N/A>
```

### Operating system

Debian

### Install method

Docker CLI

### Coral version

USB

### Network connection

Wired

### Camera make and model

Mix of Reolink, Amcrest

**Comment by NickM-27** [comment]
We've not seen this before. Somehow your database is in a bad state where the index was not created. The index was added a long while ago in previous frigate version: https://github.com/blakeblackshear/frigate/blob/dev/migrations/003_create_recordings_table.py#L41

**Comment by joeShuff** [comment]
Thats a strange one, I'm guessing the migrations are executed based on a current db version, so for some reason that migration got skipped and isn't executed every time because the db version was incremented?

I would say I'll just execute that SQL to add the index so it can be removed but its now concerning me as to what _else_ could be incorrect.

My plan of action might be, let 0.11.1 create a fresh new database, then change back to my old db and let migrations happen, then compare the db structure, do you think that might be the best approach?

**Comment by NickM-27** [comment]
That seems reasonable, I have no idea how it would have all the other things working just somehow doesn't have that index

**Comment by joeShuff** [comment]
I'll check the DB later and validate that it matches the other indexes in that migration. Thanks for linking me to that.

**Comment by joeShuff** [comment]
Turns out my DB had 0 indexes so I created the missing ones and then ran the 0.11 migrations and it is now up and running! However I'm getting a different issue now with 

> Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 954, in _bootstrap_inner
    self.run()
  File "/opt/frigate/frigate/video.py", line 215, in run
    self.start_ffmpeg_detect()
  File "/opt/frigate/frigate/video.py", line 276, in start_ffmpeg_detect
    self.ffmpeg_detect_process = start_or_restart_ffmpeg(
  File "/opt/frigate/frigate/video.py", line 133, in start_or_restart_ffmpeg
    process = sp.Popen(
  File "/usr/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.9/subprocess.py", line 1823, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'ffmpeg'

but I'll give the issues a search first and close this one off with some detail just in case others have similar.

**Comment by joeShuff** [comment]
My issue was that _somehow_ my database was out of sync, and it was missing all indexes. 

I turned the frigate container off first. Then I used [dbBrowser](https://sqlitebrowser.org/) to open up my database and compare it to a fresh 0.11 one. Then I created all the indexes that were missing to match *but* made sure to remove the ones that [the migration]( https://github.com/blakeblackshear/frigate/blob/dev/migrations/003_create_recordings_table.py#L41) was expecting to create itself and added the one it would try to remove. 

After that I created another backup and then ran frigate 0.11.1 on that modified DB, migrations succeeded and frigate booted up.
